Mark Framework interfaces as API part 2

mrtuvn opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

Mark following interfaces as API + Added strict_types

lib/internal/Magento/Framework/App/CsrfAwareActionInterface.php
lib/internal/Magento/Framework/App/DefaultPathInterface.php
lib/internal/Magento/Framework/App/DeploymentConfig/ImporterInterface.php
lib/internal/Magento/Framework/App/DeploymentConfig/ValidatorInterface.php
lib/internal/Magento/Framework/App/DeploymentConfig/Writer/FormatterInterface.php
lib/internal/Magento/Framework/App/DesignInterface.php
lib/internal/Magento/Framework/App/EnvironmentInterface.php
lib/internal/Magento/Framework/App/ExceptionHandlerInterface.php
lib/internal/Magento/Framework/App/FeedFactoryInterface.php
lib/internal/Magento/Framework/App/FeedInterface.php
lib/internal/Magento/Framework/App/HttpRequestInterface.php
lib/internal/Magento/Framework/App/ObjectManager/ConfigWriterInterface.php
lib/internal/Magento/Framework/App/PageCache/NotCacheableInterface.php
lib/internal/Magento/Framework/App/Request/ValidatorInterface.php
lib/internal/Magento/Framework/App/ResourceConnection/ConfigInterface.php
lib/internal/Magento/Framework/App/ResourceConnection/ConnectionAdapterInterface.php
lib/internal/Magento/Framework/App/Response/FileInterface.php
lib/internal/Magento/Framework/App/Response/HeaderProvider/HeaderProviderInterface.php
lib/internal/Magento/Framework/App/Response/RedirectInterface.php
lib/internal/Magento/Framework/App/Router/NoRouteHandlerInterface.php
lib/internal/Magento/Framework/App/Router/PathConfigInterface.php
lib/internal/Magento/Framework/App/RouterInterface.php
lib/internal/Magento/Framework/App/RouterListInterface.php
lib/internal/Magento/Framework/App/Rss/RssManagerInterface.php
lib/internal/Magento/Framework/App/Rss/UrlBuilderInterface.php
lib/internal/Magento/Framework/App/Scope/ValidatorInterface.php
lib/internal/Magento/Framework/App/ScopeFallbackResolverInterface.php
lib/internal/Magento/Framework/App/ScopeResolverInterface.php
lib/internal/Magento/Framework/App/ScopeTreeProviderInterface.php
lib/internal/Magento/Framework/App/ScopeValidatorInterface.php
lib/internal/Magento/Framework/App/View/Asset/MaterializationStrategy/StrategyInterface.php
lib/internal/Magento/Framework/App/View/Deployment/Version/StorageInterface.php

Also marked api for references class/interfaces

lib/internal/Magento/Framework/\App/Request/InvalidRequestException
lib/internal/Magento/Framework/Interception/ObjectManager/ConfigInterface
lib/internal/Magento/Framework/App/Response/Http
lib/internal/Magento/Framework/App/Request/Http

lib/internal/Magento/Framework/ObjectManager/RelationsInterface.php
lib/internal/Magento/Framework/ObjectManager/ConfigInterface.php
lib/internal/Magento/Framework/ObjectManager/ConfigCacheInterface.php
lib/internal/Magento/Framework/Interception/ConfigInterface.php


lib/internal/Magento/Framework/DB/LoggerInterface (by lib/internal/Magento/Framework/App/ResourceConnection/ConnectionAdapterInterface.php)

lib/internal/Magento/Framework/View/DesignInterface (by
lib/internal/Magento/Framework/App/DesignInterface.php)

### Related Pull Requests
<!-- related pull request placeholder -->
https://github.com/magento/magento2/pull/32142
https://github.com/magento/magento2/pull/32156
### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. Fixes magento/magento2#32025

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. ...
2. ...

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

mrtuvn commented:
@magento run all tests

mrtuvn commented:
Rebased 2.5-develop

mrtuvn commented:
@magento run all tests

mrtuvn commented:
Public type 'Magento\Framework\App\CsrfAwareActionInterface' references following non-public types:
Magento\Framework\App\Request\InvalidRequestException

Public type 'Magento\Framework\App\DesignInterface' references following non-public types:
Magento\Framework\View\DesignInterface

Public type 'Magento\Framework\App\EnvironmentInterface' references following non-public types:
Magento\Framework\Interception\ObjectManager\ConfigInterface

Public type 'Magento\Framework\App\ExceptionHandlerInterface' references following non-public types:
Magento\Framework\App\Response\Http
Magento\Framework\App\Request\Http

Public type 'Magento\Framework\App\ResourceConnection\ConnectionAdapterInterface' references following non-public types:
Magento\Framework\DB\LoggerInterface

mrtuvn commented:
Fix static tests
@magento run all tests

mrtuvn commented:
Fix static tests
@magento run all tests

mrtuvn commented:
@magento run Static Tests, Functional Tests B2B

mrtuvn commented:
@magento run Functional Tests B2B, Static Tests

mrtuvn commented:
@magento run Functional Tests B2B

eduard13 commented:
@mrtuvn instead of marking the `api` classes that are failing on Static Tests, it's better to take a quick look, maybe they were already mentioned in scope of another PRs. For example, here you are marking as api:

- lib/internal/Magento/Framework/DB/LoggerInterface - but this was already marked in another your PR #32156

This will cause a lot of conflicts while merging them.

Please add all the PRs that you find useful for your PR, as related.

Please let me know if any questions arise.

mrtuvn commented:
Fixed issue related as your suggestion

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run Static Tests, Unit Tests

mrtuvn commented:
@magento run Static Tests, Unit Tests

mrtuvn commented:
@magento run Static Tests

mrtuvn commented:
@magento run Database Compare, Functional Tests B2B, Functional Tests CE, Functional Tests EE, Integration Tests, Magento Health Index, Sample Data Tests B2B, Sample Data Tests CE, Sample Data Tests EE, Semantic Version Checker, WebAPI Tests

eduard13 commented:
@magento run Unit Tests, Functional Tests EE, Functional Tests CE, Functional Tests B2B

eduard13 commented:
@magento run Functional Tests EE, Functional Tests CE, Functional Tests B2B

mrtuvn commented:
@magento run all tests

eduard13 commented:
@magento run Functional Tests EE, WebAPI Tests

eduard13 commented on lib/internal/Magento/Framework/App/Response/Http.php:
```diff
@@ -131,7 +133,7 @@ public function sendVary()
     public function setPublicHeaders($ttl)
     {
-        if ($ttl < 0 || !preg_match('/^[0-9]+$/', $ttl)) {
+        if ($ttl < 0 || !$ttl) {
```
Hi @mrtuvn, any info of why the following logic has been changed?

mrtuvn replied:
After add strict type i have found error here when second argument of preg_match required type string. But we passed TTL with int value. So i think we don't have to check with preg_match

eduard13 replied:
I think the main reason of why it was failing, is the following test that passes a `null` value here https://github.com/magento/magento2/blob/45f27d64cbb475a4af4d27e14d312653823083e1/lib/internal/Magento/Framework/App/Test/Unit/Response/HttpTest.php#L231

But preg_match's 2nd argument should be a string value.

But now, we allow passing string values as parameter, as we still don't have any restrictions, which shouldn't happen:
- asd
- test123
- =1
- etc

We should avoid that, and should only have int values.

eduard13 reviewed (changes requested):
Requesting changes

mrtuvn commented:
What should i need to change ?

mrtuvn commented on lib/internal/Magento/Framework/App/Response/Http.php:
```diff
@@ -131,7 +133,7 @@ public function sendVary()
+        if ($ttl < 0 || !preg_match('/^[0-9]+$/', (string) $ttl)) {
             throw new \InvalidArgumentException('Time to live is a mandatory parameter for set public headers');
         }
         $this->setHeader('pragma', 'cache', true);
```
Added compulsive type here for input `$ttl` . we have explicit type pass to preg_match
CC: @eduard13

eduard13 replied:
But, have you tried to update the Unit Test that I mentioned, to see if that's the reason of why the test was previously failing (referencing the comment above)?

mrtuvn replied:
Test file still remain i don't change the null test case! Only change in origin file. Re-run unit test passed in my local

eduard13 commented:
@magento run all tests

eduard13 commented:
@magento run Functional Tests B2B

eduard13 commented:
@magento run Functional Tests B2B

mrtuvn commented:
@magento run Functional Tests B2B

eduard13 commented on lib/internal/Magento/Framework/App/Response/Http.php:
```diff
@@ -131,7 +133,7 @@ public function sendVary()
     public function setPublicHeaders($ttl)
     {
-        if ($ttl < 0 || !preg_match('/^[0-9]+$/', $ttl)) {
+        if ($ttl < 0 || !preg_match('/^[0-9]+$/', (string) $ttl)) {
```
@mrtuvn after discussing this change internally, decided that it makes sense to get rid of this `preg_match`. This way, we won't need the `string` casting.
```suggestion
        if ($ttl < 0 || !is_numeric($ttl)) {
```

mrtuvn replied:
Thanks you just corrected it again as suggestion

mrtuvn commented:
@mgento run all tests

eduard13 commented:
@magento run all tests

mrtuvn commented:
@magento run Database Compare

eduard13 commented:
@magento run Database Compare

eduard13 commented:
@magento run Database Compare

eduard13 reviewed (approved):
Don't think that Database Compare failed builds are somehow related with these changes.

magento-engcom-team commented:
Hi @eduard13, thank you for the review.
[ENGCOM-8920](https://jira.corp.magento.com/browse/ENGCOM-8920) has been created to process this Pull Request

mrtuvn commented:
@magento run Database Compare

mrtuvn commented:
Re-targeted PR branch to 2.4-develop as internal discussed with @gabrieldagama

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run Functional Tests EE

gabrieldagama commented on lib/internal/Magento/Framework/App/Response/Http.php:
```diff
@@ -131,7 +133,7 @@ public function sendVary()
     public function setPublicHeaders($ttl)
     {
-        if ($ttl < 0 || !preg_match('/^[0-9]+$/', $ttl)) {
+        if ($ttl < 0 || !is_numeric($ttl)) {
```
We are not looking for code changes with these issues, please revert as it was.

mrtuvn replied:
This change introduce since i added strict_type at the first line of fine! This change pass test too should i revert strict_type in only this

gabrieldagama replied:
In this case, let's create a separate PR to add the strict type and the code change. On this scope, we are looking just for API marking.

mrtuvn replied:
updating it

mrtuvn commented:
@magento run all tests

mrtuvn committed:
Mark Framework interfaces as API part 2


Update


hollymolly


up


update


update types


fix fail tests


update strict type for preg_match 2rd argument


Update


revert origin code


resolve conflicts

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run Functional Tests B2B

mrtuvn commented:
@gabrieldagama all green tests and ready!

Commit:
Merge branch 'api-changes-develop' into api-interfaces-mark-2

gabrieldagama commented:
@magento import code to https://github.com/magento-engcom/magento2ce

magento-engcom-team commented:
@gabrieldagama the branch with code successfully imported into`magento-engcom/magento2ce` repository. Branch name: `imported-magento-magento2-32140`.
